features/eager_activity/non_remote_activities_worker/feature.py

```diff
@@ -0,0 +1,42 @@
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.exceptions import ActivityError, TimeoutError, TimeoutType
+from temporalio.worker import WorkerConfig
+
+from harness.python.feature import register_feature
+
+
+@workflow.defn
+class Workflow:
+    @workflow.run
+    async def run(self) -> None:
+        # Run a workflow that schedules a single activity with short schedule-to-close timeout
+        try:
+            await workflow.execute_activity(
+                dummy,
+                # Pick a long enough timeout for busy CI but not too long to get feedback quickly
+                schedule_to_close_timeout=timedelta(seconds=3),
+            )
+            raise RuntimeError("Expected activity to time out")
+        except ActivityError as e:
+            # Catch activity failure in the workflow, check that it is caused by schedule-to-start timeout
+            if not isinstance(e.cause, TimeoutError):
+                raise e
+
+            assert isinstance(e.cause, TimeoutError)
```

Comment:
Already effectively done by the conditional above

Reply:
I needed this one for mypy to infer the type, can remove the condition above.